pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -341,7 +385,9 @@ func (a *Assignment) findFlavorForResourceGroup(
 
 	// We will only check against the flavors' labels for the resource.
 	selector := flavorSelector(spec, rg.LabelKeys)
-	for _, flvQuotas := range rg.Flavors {
+	lastScheduledFlavorIdx := 0
```

Comment:
Within a resource group, all the resources get assigned the same flavor.
So at this point you can find the last flavor used for any of the resources in this resource group, and start the iteration from there.

It would be the same behavior you have below, but I think it will be easier to read.